backend/src/schema/resolvers/emails.spec.js

```diff
@@ -0,0 +1,283 @@
+            await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+              data: { VerifyEmailAddress: null },
+              errors: [{ message: 'Invalid nonce or no email address found.' }],
+            })
+          })
+        })
+
+        describe('and the `EmailAddressRequest` belongs to the authenticated user', () => {
+          beforeEach(async () => {
+            await emailAddress.relateTo(user, 'belongsTo')
+          })
+
+          it('adds `verifiedAt`', async () => {
+            await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+              data: {
+                VerifyEmailAddress: {
+                  email: 'to-be-verified@example.org',
+                  verifiedAt: expect.any(String),
+                  createdAt: expect.any(String),
+                },
+              },
+              errors: undefined,
+            })
+          })
+
+          it('connects the new `EmailAddress` as PRIMARY', async () => {
+            await mutate({ mutation, variables })
+            const result = await neode.cypher(`
+            MATCH(u:User)-[:PRIMARY_EMAIL]->(e:EmailAddress {email: "to-be-verified@example.org"})
+            MATCH(u:User)<-[:BELONGS_TO]-(:EmailAddress {email: "user@example.org"})
+            RETURN e
+          `)
+            const email = neode.hydrateFirst(result, 'e', neode.model('EmailAddress'))
+            await expect(email.toJson()).resolves.toMatchObject({
+              email: 'to-be-verified@example.org',
+            })
+          })
+
+          it('removes previous PRIMARY relationship', async () => {
+            const cypherStatement = `
+            MATCH(u:User)-[:PRIMARY_EMAIL]->(e:EmailAddress {email: "user@example.org"})
+            RETURN e
+          `
+            let result = await neode.cypher(cypherStatement)
+            let email = neode.hydrateFirst(result, 'e', neode.model('EmailAddress'))
+            await expect(email.toJson()).resolves.toMatchObject({
+              email: 'user@example.org',
+            })
+            await mutate({ mutation, variables })
+            result = await neode.cypher(cypherStatement)
+            email = neode.hydrateFirst(result, 'e', neode.model('EmailAddress'))
+            await expect(email).toBe(false)
+          })
+
+          describe('Edge case: In the meantime someone created an `EmailAddress` node with the given email', () => {
+            beforeEach(async () => {
+              await factory.create('EmailAddress', { email: 'to-be-verified@example.org' })
+            })
+
+            it('throws UserInputError because of unique constraints', async () => {
+              await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+                data: { VerifyEmailAddress: null },
+                errors: [{ message: 'A user account with this email already exists.' }],
+              })
+            })
+          })
+        })
+      })
+    })
+  })
+})
```

Comment:
Hard to believe, but I test drove the backend :wink: 

For the frontend I added tests after the implementation because at the time I had hopes to find a solution with less page components and redirects. In the end I figured that solution to be the least complex though.

Reply:
awesome :clap: